components/selector/__init__.py

```diff
@@ -27,7 +28,7 @@ class Selector(SchedulerComponent):
     information is statically determined.
     """
     collector: Collector
-
+    static_Env: ClassVar[Env] = Env()
```

Comment:
Also, not to be nitpicky, but Python has a very well-defined set of how naming conventions should work. (This is why I use PyCharm instead of MVC... it tells me all the warnings that violate Python standards and I don't think that MVC does.)

Just an FYI: variable names in Python (unless they're constants) are supposed to always be written in snake case, which is lowercase with underscores, so this should be called `static_env` instead of `static_Env`.

In fact, something I didn't mention, but we may as well go and make this variable "private" to indicate that it shouldn't be accessed outside of the class since we'll only need it in here and nowhere else. That means starting the name with an underscore.

So let's just declare it as:
```
_env: ClassVar[Env] = Env()
```
which is right.